extensions/wikia/ThemeDesigner/ThemeSettings.class.php

```diff
@@ -117,120 +139,168 @@ public function getHistory() {
+			$temp_file->delete( '' );
 
-		// Verify wordmark length ( CONN-116 )
-		if ( !empty( $settings[ 'wordmark-text' ] ) ) {
-			$settings[ 'wordmark-text' ] = trim( $settings[ 'wordmark-text' ] );
-		}
+			$settings["{$name}"] = $file->getURL();
+			$settings["{$name}-name"] = $file->getName();
 
-		if ( empty( $settings[ 'wordmark-text' ] ) ) {
-			// Do not save wordmark if its empty.
-			unset( $settings[ 'wordmark-text' ] );
-		} else {
-			if ( mb_strlen( $settings[ 'wordmark-text' ] ) > 50 ) {
-				$settings[ 'wordmark-text' ] = mb_substr( $settings[ 'wordmark-text' ], 0, 50 );
+			if ( $setDimensions ) {
+				$settings["${name}-height"] = $file->getHeight();
+				$settings["${name}-width"] = $file->getWidth();
 			}
-		}
 
-		if ( isset( $settings['favicon-image-name'] ) && strpos( $settings['favicon-image-name'], 'Temp_file_' ) === 0 ) {
-			$temp_file = new LocalFile( Title::newFromText( $settings['favicon-image-name'], 6 ), RepoGroup::singleton()->getLocalRepo() );
-			$file = new LocalFile( Title::newFromText( self::FaviconImageName, 6 ), RepoGroup::singleton()->getLocalRepo() );
-			$file->upload( $temp_file->getPath(), '', '' );
-			$temp_file->delete( '' );
-			Wikia::invalidateFavicon();
+			if ( !empty( $previewThumbnailDimensions ) ) {
+				$imageServing = new ImageServing(
+					null,
+					$previewThumbnailDimensions['width'],
+					[
+						'w' => $previewThumbnailDimensions['width'],
+						'h' => $previewThumbnailDimensions['height']
+					]
+				);
+				$settings["user-{$name}"] = $file->getURL();
+				$settings["user-{$name}-thumb"] = wfReplaceImageServer(
+					$file->getThumbUrl(
+						$imageServing->getCut(
+							$file->getWidth(),
+							$file->getHeight(),
+							'origin'
+						) . '-' . $file->getName()
+					)
+				);
+			}
 
-			$settings['favicon-image-url'] = $file->getURL();
-			$settings['favicon-image-name'] = $file->getName();
+			if ( is_callable( $callback ) ) {
+				$callback();
+			}
 
 			$file->repo->forceMaster();
 			$history = $file->getHistory( 1 );
+
 			if ( count( $history ) == 1 ) {
-				$oldFaviconFile = array( 'url' => $history[0]->getURL(), 'name' => $history[0]->getArchiveName() );
+				return [ 'url' => $history[0]->getURL(), 'name' => $history[0]->getArchiveName() ];
 			}
 		}
 
-		if ( isset( $settings['wordmark-image-name'] ) && strpos( $settings['wordmark-image-name'], 'Temp_file_' ) === 0 ) {
-			$temp_file = new LocalFile( Title::newFromText( $settings['wordmark-image-name'], 6 ), RepoGroup::singleton()->getLocalRepo() );
-			$file = new LocalFile( Title::newFromText( self::WordmarkImageName, 6 ), RepoGroup::singleton()->getLocalRepo() );
-			$file->upload( $temp_file->getPath(), '', '' );
-			$temp_file->delete( '' );
+		return null;
+	}
 
-			$settings['wordmark-image-url'] = $file->getURL();
-			$settings['wordmark-image-name'] = $file->getName();
+	public function saveSettings( $settings, $cityId = null ) {
+		global $wgCityId, $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)